zipline/finance/performance/period.py

```diff
@@ -203,9 +200,12 @@ def calculate_performance(self):
 
     def record_order(self, order):
         if self.keep_orders:
-            dt_orders = self.orders_by_modified[order.dt]
-            if order.id in dt_orders:
-                del dt_orders[order.id]
+            try:
+                dt_orders = self.orders_by_modified[order.dt]
+                if order.id in dt_orders:
+                    del dt_orders[order.id]
```

Comment:
The only reason I can think of that this might matter is if we have a custom __hash__ or something strange like that.